src/components/createPostModal/index.js

```diff
@@ -11,18 +12,24 @@ const CreatePostModal = () => {
 
     const [message, setMessage] = useState(null)
     const [text, setText] = useState('')
+    const [isLoading, setIsLoading] = useState(false)
 
     const onChange = (e) => {
         setText(e.target.value)
     }
 
-    const onSubmit = () => {
-        setMessage('Submit button was clicked! Closing modal in 2 seconds...')
-
-        setTimeout(() => {
-            setMessage(null)
-            closeModal()
-        }, 2000)
+    const onSubmit = async () => {
+        setIsLoading(true)
+        try {
+            await createPost(text)
```

Comment:
Since we are doing a `.then()` we don't need `async/await` here - its usually either or